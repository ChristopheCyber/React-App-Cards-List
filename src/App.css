.bigLow {
  font-family: "Bigelow Rules";
  font-size: 30px;
  text-shadow: 2px 8px 6px rgb(228, 43, 43);
  font-style: normal;
  /*text-decoration: underline;*/
  text-decoration-thickness: 2px;
  color: #4bccdd;
}
.bigLow-search-filtered {
  color: rgb(120, 245, 37);
}
.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: rgb(18, 216, 28);
}

.App-link {
  color: #61dafb;
}
/**/
@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
/*FONT FACE*/
/* latin BigeLow */
@font-face {
  font-family: "Bigelow Rules";
  font-style: normal;
  font-weight: 400;
  src: local("Bigelow Rules Regular"), local("BigelowRules-Regular"),
    url(https://fonts.gstatic.com/s/bigelowrules/v8/RrQWboly8iR_I3KWSzeRuN0zf4AIHg.woff2)
      format("woff2");
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA,
    U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215,
    U+FEFF, U+FFFD;
}
