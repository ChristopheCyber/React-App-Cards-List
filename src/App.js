import React from 'react';
import './App.css';
import { CardList } from './components/card-list/card-list.component.jsx';


class App extends React.Component {
    constructor(){
    super();
    this.state={//Array cards
      cards:[]
    }
  }
  componentDidMount(){
    fetch("https://jsonplaceholder.typicode.com/users")
    .then ( resp1 => resp1.json() )
    .then ( users => this.setState( {cards : users} ) )
  }
  render() {
    return (
      <div className="App">
        <CardList cardsProp1={this.state.cards}>
          {/* children of CardList component 
          <h2>CardList children here </h2>
          {
            this.state.cards.map( cardElt => 
              <h1 key={cardElt.id}> {cardElt.name} </h1> )
          }
          */}
        </CardList> 
      </div>
    )
  }
}
export default App;
