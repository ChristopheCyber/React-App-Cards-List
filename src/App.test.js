import React from 'react';
// import { render } from '@testing-library/react';
import App from './App';

//ENZYME fncts: ( we don't use MOUNT and RENDER here )
import { shallow } from "enzyme";

/*
test('/Cards Selection Board/ text', () => {
  expect.assertions(1);
  const { getByText } = render(<App />);
  const linkElement = getByText(/Cards Selection Board/i);
  expect(linkElement).toBeInTheDocument();
});
 */

describe('Group Script App.test.I: ', () => {
  it('Test App.test.I.1; Snapshot testing:', () => {
      expect.assertions(1);
      // testing if shallow rendering correctly:
      expect(shallow( <App /> )).toMatchSnapshot();
  });
});

