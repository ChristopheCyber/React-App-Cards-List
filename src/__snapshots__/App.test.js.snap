// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group Script App.test.I:  Test App.test.I.1; Snapshot testing: 1`] = `
<div
  className="App img-class"
>
  <h1>
     Cards Selection Board 
  </h1>
  <label>
    <i
      className="bigLow js-grid"
    >
      Search in cards titles -&gt;
    </i>
  </label>
  <SearchBox
    handleChangeFct={[Function]}
    propPlaceHolder="Type text for Searching"
  />
  <label>
    <i
      className="bigLow bigLow-search"
    >
      Â  
       ( 0 card in list now ) 
    </i>
  </label>
  <CardList
    cardsProp1={Array []}
    cardsPropImg="img-class"
  />
  <div
    id="tsparticles"
  >
    <h
      height="100%"
      id="tsparticles"
      options={
        Object {
          "detectRetina": true,
          "fpsLimit": 60,
          "interactivity": Object {
            "detectsOn": "window",
            "events": Object {
              "onClick": Object {
                "enable": true,
                "mode": "grab",
              },
              "onHover": Object {
                "enable": true,
                "mode": "push",
              },
              "resize": true,
            },
            "modes": Object {
              "bubble": Object {
                "distance": 400,
                "duration": 2,
                "opacity": 0.8,
                "size": 40,
              },
              "push": Object {
                "quantity": 4,
              },
              "repulse": Object {
                "distance": 50,
                "duration": 0.6,
              },
            },
          },
          "particles": Object {
            "collisions": Object {
              "enable": true,
            },
            "color": Object {
              "animation": Object {
                "enable": true,
                "speed": 40,
                "sync": true,
              },
              "value": "#ff0000",
            },
            "links": Object {
              "color": "#ffffff",
              "distance": 150,
              "enable": true,
              "opacity": 0,
              "width": 1,
            },
            "move": Object {
              "direction": "random",
              "enable": true,
              "outMode": "out",
              "random": true,
              "speed": 2,
              "straight": true,
            },
            "number": Object {
              "density": Object {
                "enable": true,
                "value_area": 1000,
              },
              "value": 80,
            },
            "opacity": Object {
              "value": 0.5,
            },
            "shape": Object {
              "type": "circle",
            },
            "size": Object {
              "random": true,
              "value": 7,
            },
          },
        }
      }
      style={Object {}}
      width="100%"
    />
  </div>
</div>
`;
